### DISCUSSION TITLE
Update system Extensions

### DISCUSSION THREAD

**Discussion by inck86** [vladmandic/sdnext#794] [General]
I noticed that when checking for updates, system extensions are not checked. For example, ControlNet is a system one, it is not checked for updates, how will it be implemented if its update appears? using git pull WebUI automatically?
![image](https://user-images.githubusercontent.com/119116007/236647163-c5138448-8f92-464c-819e-935ca10adba2.png)

**Comment by vladmandic** [comment]
For now, all built in extensions are updated automatically. I plan to redo extension manager completely and then you'll have the control.

**Reply by scpedicini** [reply]
@vladmandic I just did a fresh install of SD.Next and noticed that a lot of the System extensions have the "Update" button enabled, e.g. `sd-webui-controlnet`, `sd-extension-chainner`, `sd-extension-system-info`, etc. Would there be any reason not upgrade? Can I just safely press the "Update all installed" button?

EDIT: I pressed `Update` on a few of them. Other than changing the button label to "Updating..." - Log output shows:
```
10:43:20-389223 DEBUG    Extensions update start: sd-extension-system-info 1841cf7627d3991d7ba3dfe8b86a14b18f78f933 1696946888
10:43:20-666883 DEBUG    Extensions update finish: sd-extension-system-info 1841cf7627d3991d7ba3dfe8b86a14b18f78f933 1696946888
```

However, refreshing the entire page the "Update" button is still available for this extension.
